posts/Yoann/2017-11-27-introducing-hibernate-search-jsr352-mass-indexing-job.adoc

```diff
@@ -0,0 +1,158 @@
+= Introducing the Hibernate Search JSR-352 mass indexing job
+Yoann Rodière
+:awestruct-tags: [ "Hibernate Search", "JSR-352" ]
+:awestruct-layout: blog-post
+---
+
+Hibernate Search 5.9 (still in beta) features a new integration to JSR-352 runtimes,
+providing a new implementation of mass indexing (indexing a high volume of entities)
+as a JSR-352 job.
+
+Why is it nice? How to use it? Let's find out!
+
+== What's JSR-352?
+
+"JSR 352: Batch Applications for the Java Platform" is a Java specification
+defining interfaces and mechanisms to execute batch processes ("jobs"),
+i.e. processes that involve applying the same operation over and over again to a large number of items.
+
+In JSR-352:
+
+* The job definition is provided as an XML file and a few Java classes, called batch artifacts
+  (in our case, the definition is provided by Hibernate Search).
+* The JSR-352 runtime avoids boilerplate code by managing
+  threading, transactions, the execution flow (calling methods on batch artifacts), ...
+* And all that's left is configuring the infrastructure
+  (in our case that's your responsibility, as the Hibernate Search user).
+
+== What's better than the old mass indexer?
+
+Ever tried to reindex 100,000 entities with the old mass indexer,
+only to experience some network or disk error at the 99,001th entity?
+Then you have to restart from the very beginning,
+even though the first 99,000 entities were indexed just fine...
+
+With the JSR-352 mass indexing job, the job will simply resume from the last checkpoint,
+probably somewhere around the 95,000th entity.
+
+There are other advantages (such as being able to pool resources between multiple, unrelated background jobs),
+but this one really is why we think JSR-352 makes sense.
```

Comment:
`this one really is why we think` doesn't flow very well in English. "why" isn't the same as "the reason for" - as in latin languages..

Reply:
Ok, I'll change that, but... I frequently see "this is why + subject + verb" or similar phrasing... Is there something special here that makes it incorrect?

Reply:
it's indeed slightly different. The "why" is to answer the question, so "this is why xyz" is similar to "this is to answer the question you might have about xyz" but you can't say "xyz is the why"